packages/datadog_grpc_interceptor/test/datadog_grpc_interceptor_test.dart

```diff
@@ -39,77 +39,246 @@ void main() {
+
+      expect(attributes['grpc.method'], '/helloworld.Greeter/SayHello');
+
+      verify(
+          () => mockRum.stopResourceLoading(key, 200, RumResourceType.native));
+    });
+
+    test('Interceptor calls send tracing attributes', () async {
+      when(() => mockDatadog.isFirstPartyHost(any())).thenReturn(true);
+
+      final interceptor = DatadogGrpcInterceptor(mockDatadog, channel);
+
+      final stub = GreeterClient(channel, interceptors: [interceptor]);
+
+      await stub.sayHello(HelloRequest(name: 'test'));
+
+      final captures = verify(() => mockRum.startResourceLoading(
+          captureAny(),
+          RumHttpMethod.get,
+          'http://localhost:$port/helloworld.Greeter/SayHello',
+          captureAny())).captured;
+      final attributes = captures[1];
+      expect(attributes['_dd.trace_id'], isNotNull);
+      expect(BigInt.tryParse(attributes['_dd.trace_id'] as String), isNotNull);
+      expect(attributes['_dd.span_id'], isNotNull);
+      expect(BigInt.tryParse(attributes['_dd.span_id'] as String), isNotNull);
+    });
+
+    test(
+        'Interceptor calls do not send tracing attributes when shouldSample returns false',
+        () async {
+      when(() => mockDatadog.isFirstPartyHost(any())).thenReturn(true);
+      when(() => mockRum.shouldSampleTrace()).thenReturn(false);
+
+      final interceptor = DatadogGrpcInterceptor(mockDatadog, channel);
+
+      final stub = GreeterClient(channel, interceptors: [interceptor]);
+
+      await stub.sayHello(HelloRequest(name: 'test'));
+
+      final captures = verify(() => mockRum.startResourceLoading(
+          captureAny(),
+          RumHttpMethod.get,
+          'http://localhost:$port/helloworld.Greeter/SayHello',
+          captureAny())).captured;
+      final attributes = captures[1];
+      expect(attributes['_dd.trace_id'], isNull);
+      expect(attributes['_dd.span_id'], isNull);
+    });
+
+    test('Interceptor passes on proper metadata', () async {
+      when(() => mockDatadog.isFirstPartyHost(any())).thenReturn(true);
+
+      final interceptor = DatadogGrpcInterceptor(mockDatadog, channel);
+
+      final stub = GreeterClient(channel, interceptors: [interceptor]);
+
+      await stub.sayHello(HelloRequest(name: 'test'));
+
+      expect(loggingService.calls.length, 1);
+      final call = loggingService.calls[0];
+      expect(call.clientMetadata!['x-datadog-trace-id'], isNotNull);
+      expect(
+          BigInt.tryParse(call.clientMetadata!['x-datadog-trace-id'] as String),
+          isNotNull);
+      expect(call.clientMetadata!['x-datadog-parent-id'], isNotNull);
+      expect(
+          BigInt.tryParse(
+              call.clientMetadata!['x-datadog-parent-id'] as String),
+          isNotNull);
+      expect(call.clientMetadata!['x-datadog-origin'], 'rum');
+      expect(call.clientMetadata!['x-datadog-sampling-priority'], '1');
+    });
+
+    test('Interceptor does not send traces for when shouldSample returns false',
```

Comment:
seems test name has a mistake?

Reply:
Is it just that extra "for" or am I missing something else?

Reply:
yes, also maybe it should say `traces metadata` by looking on the title of the previous test.